méli-mélo/2023-10-clipboard/index.html

```diff
@@ -0,0 +1,386 @@
+---
+title: Copy to clipboard
+dateModified: 2023-09-07
+description: "Create a button and copies the indicated content to the clipboard"
+lang: en
+css: 
+- css/clipboard.css
+script: 
+- js/clipboard.js
+---
+
+
+<p>Create a button and copies the indicated content to the clipboard</p>
+
+<h2>GCWeb implementation plan</h2>
+
+{% assign implPlan = site.pages | where: "output", "false" | where: "componentName", "2021-10-clipboard" | first %}
+
+<ul>
+{% for deliverable in implPlan.implementationPlan %}
+	<li>{{ deliverable.due }} - {{ deliverable.what }}</li>
+{% endfor %}
+</ul>
+
+<p>Todo and for future consideration</p>
+<ul>
+{% for todo in implPlan.todos %}
+	<li>{{ todo }}</li>
+{% endfor %}
+</ul>
+
+<p>Sponsor: {{ implPlan.sponsor }}</p>
+
+
+    <section>
+        <h2>Purpose</h2>
+        <p>This feature provides an interface to copy content to the clipboard.</p>
+        </section>
+    <section>
+        <h2>How to use</h2>
+        <ol>
+        <li>Apply <code>class="wb-clipboard"</code> to an element on the page to load the component.</li>
+        </ol>
+        </section>
+<!-- Copy to Clipboard Example 1 -->
+<h2 id="examples">How to implement</h2>
+<h3>Default</h3>
+<div class="wb-clipboard"><p>Lorem <a href="https://wet-boew.github.io/wet-boew-styleguide/index-en.html">ipsum dolor sit</a> amet, consectetur adipiscing elit. Vivamus luctus sem eros, consectetur posuere velit blandit vitae. Suspendisse quam nibh, suscipit vel turpis non, <strong>dignissim pellentesque</strong> ligula. Pellentesque et aliquam enim, vel convallis justo.</p>
+    <p>Nam feugiat:</p>
+    <ul>
+        <li>eros eget tempor lacinia, </li>
+        <li>sapien purus imperdiet nibh, </li>
+        <li>nec rhoncus justo tellus euismod ex. </li>
+    </ul>
+    <p><pre><code>&lt;span class=&quot;test&quot;&gt;test this&lt;/span&gt;</code></pre></p>
+    <div class="row hidden">
+        <div class="col-sm-2">
+            <img src="../../images/playbutton.png" alt="" class="img-responsive">
+        </div>
+    </div>
+</div>
```

Comment:
**Suggestion:** This is just my opinion, and is related to usability only, but I think the button should always remain into the boundaries of the text it is meant to copy. Otherwise, it isn't clear visually if the text to copy is the one above or under the button, and determining what is the scope of the copying. Quick workaround I can think of would be in having an introductory text clearly mentioning that the following text can be copied in one click, or having the label of the button state "Copy text above" by using custom button... but anyway jsut food for thoughts.